coalib/parsing/InvalidFilterException.py

```diff
@@ -1,2 +1,8 @@
 class InvalidFilterException(Exception):
-    pass
+    def __init__(self, filter_name=None, available_filters=None):
```

Comment:
Please remove default values. Otherwise we'll need to add conditions to check that.
We can safely assume that if one wants to raise `InvalidFilterException`, then there is definitely a `filter_name` which is invalid. So, it can't be defaulted to `None`. And to check whether a filter is valid or not, there has to be a list of available filters. So, that parameter can't be `None` as well.

Reply:
@aashutoshrathi @shreyans800755 We should also think about if `FilterHelper.available_filters` ever change during runtime. If not, that parameter becomes obsolete

Reply:
@aashutoshrathi @shreyans800755 https://github.com/coala/coala/pull/4821/files/f47beedaade7a74a7cf71a7a31283522006dcc43#r150385623 above is misleading ;) The actual question is if there will be ever other `available_filters` than global `FilterHelper.available_filters`